src/headline_grabber/pipeline_steps/scrape_text.py

```diff
@@ -31,14 +31,14 @@ def _filter_results(self, x: Headline) -> bool:
             and len(" ".join([x.title, x.description])) > min_content_length
         )
 
-    def _parse_headline(self, tag: Tag, page_selectors: PageSelectors) -> Headline:
+    def _parse_headline(self, tag: Tag, page_selectors: PageSelectors, url: str = "") -> Headline:
         link_selector = page_selectors.link
         title_selector = page_selectors.title
         description_selector = page_selectors.description
 
         return Headline(
             link=(
-                tag.find(link_selector.tag)[link_selector.identifier]
+                (url + tag.find(link_selector.tag)[link_selector.identifier])
```

Comment:
Rather than add another config, better would probably be to check whether the link starts with the base url, and if it doesn't, then add it. Otherwise, just use the raw link.

Reply:
It will not break any link for the other sites. We have the url parameter to control that. If we don't put any specific root url, url = empty string.